core/main/src/service/apps/app_events.rs

```diff
@@ -259,13 +256,43 @@ impl AppEvents {
         }
     }
 
+    fn get_rpc_v2_result(event: &str, input: Value) -> Value {
+        // FIXME: This is a temporary hack to get the event field name from the event string.
+        // We'll need to extract this from the firebolt schema when it's available.
+
+        debug!("get_rpc_v2_result: event={}, input={:?}", event, input);
+
+        let mut result = input.clone();
+        let method = event.split('.').collect::<Vec<&str>>()[1];
+
+        if let Some(start) = method.find("on") {
+            if let Some(end) = method.find("Changed") {
+                if start < end {
+                    let field_name = &method[start + 2..end];
+                    let field_name_cased =
+                        format!("{}{}", &field_name[0..1].to_lowercase(), &field_name[1..]);
+                    result = json!({field_name_cased: input});
+                }
+            }
+        }
+
+        debug!("get_rpc_v2_result: result={:?}", result);
+        result
+    }
+
     pub async fn send_event(listener: &EventListener, data: &Value) {
```

Comment:
Modified to fix RPCv2 support for events that don't have rules defined (e.g. handler-based).